[gradle] print out task name instead of 'gradle' step

att55 opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

To prevent printing out multiple 'gradle step' in summary, `step_name` option is prepared.

```
+------+-------------+-------------+
|         fastlane summary         |
+------+-------------+-------------+
| Step | Action      | Time (in s) |
+------+-------------+-------------+
| 1    | gradle      | 10          |
| 2    | gradle      | 8           |
| 3    | gradle      | 15          |
+------+-------------+-------------+
```

But, I think `task name` is used as `step_name` instead of 'gradle' in most cases.
In these case, the code in Fastfile is a little **redundant** as below... 💧 

```
gradle(task: 'assmebleDebug', step_name: 'assembleDebug')
gradle(task: 'bundleDebug', step_name: 'bundleDebug')
```

So, I think `task name` is better than 'gradle' to be printed out as default.

```
+------+----------------+-------------+
|         fastlane summary            |
+------+----------------+-------------+
| Step | Action         | Time (in s) |
+------+----------------+-------------+
| 1    | assmebleDebug  | 10          |
| 2    | bundleDebug    | 8           |
+------+----------------+-------------+
```

### Description
<!-- Describe your changes in detail. -->
<!-- Please describe in detail how you tested your changes. -->

In case gradle action is called, `task name` is printed out instead of 'gradle step'.
If `step_name` is passed, `step_name` is printed out.

att55 committed:
use task name instead of gradle step

rm unnecessary white space

att55 committed:
fix comments

joshdholtz reviewed (changes requested):
I actually _really_ like how this looks but forcing the runner to know about the gradle action is not something we'd like to do 🤔  The runner shouldn't have anything action specific. 

I think it would be cool if there was a way the **any** action could proide this 😇 I don't think know how easy this would be with the current implementation of actions but it would be nice to look into if possible 👌

att55 commented:
@joshdholtz Thank you for your review!

> The runner shouldn't have anything action specific.

You are completely right, I think the runner should be global module too...
So, I will try to find another way, which the change effects only gradle action or any action could provide this (as you written).

att55 commented:
I will add the title to `WIP` 🙏

att55 committed:
implement in gradle action

rm unnecessary code

fix to add array method

fix for lint

fix arguments

delete step_name option

arrange names

att55 committed:
add unit tests

fix for lint

fix for lint

fix for lint

fix for lint

att55 commented:
@joshdholtz @janpio I added the parameter `print_task_name_in_summary` to gradle action not to effect the runner. How is this implementation? 

- before diff
https://github.com/fastlane/fastlane/pull/15526/commits/d318d1e0e318082ae79b34e7621a3969438ba3f7

att55 commented:
@janpio Please remove `status: wip` label 🙏

att55 commented:
@joshdholtz Please review when you can 🙏

janpio commented:
Rebase onto master should get rid of the last failing CI.

att55 committed:
modify comment

att55 commented:
@janpio Got rid of the last failure 🙆‍♂

att55 commented:
@joshdholtz Please review this PR when you can 🙏

joshdholtz commented:
Aye aye! Will review first thing tomorrow AM

att55 commented:
@joshdholtz Thank you for your reply ! That would be fine 🙆‍♂

joshdholtz reviewed (commented):
@att55 This is great improvement! This actually inspired me to add a commit to make this a little more powerful and flexible 😊 Commit incoming in a bit! If you are able to review/test this that would be 💯

Commit:
Allow passing of configuration/arguments into step_text for customzing of displayable step name in action

joshdholtz commented:
@googlebot I consent.

joshdholtz reviewed (approved):
I have some commits in here but this looks 💯 to me 😊

snatchev commented on fastlane/lib/fastlane/actions/gradle.rb:
```diff
@@ -94,6 +94,20 @@ def self.run(params)
         return result
       end
 
+      def self.step_text(params)
```
Is there any way we could the task name (and others) without changing the method signatures?

joshdholtz replied:
Not without a rearchitecture 😬 Actions and their methods are execute as class methods and not as instance methods so there is no shared state that `step_text` can get 😢

snatchev replied:
Understood.

snatchev commented:
I would recommend we do not introduce a new option for this, but instead we just adapt this as the new behavior for gradle actions.

I don't love the fact that this change forces the `step_text` on the other runners to change their signatures. An alternative would be to check the arity of `step_text` and not pass `params` if it's arity is 0. This seems a bit hacky and magical, but the alternative would require re-architecting these methods to not be class methods which also has some downsides. Any thoughts @att55 @joshdholtz ?

joshdholtz commented:
@snatchev Checking if it could take a param or not did cross my mind originally when reworking some of this but it seemed less buggy and more straight forward to just adjust those few actions that needed the change. But now that I think about it, that might be needed for any plugins that use `step_text` 🤔 So yeeaaaahhhh... might have to make that change to check for arity.

att55 commented:
@snatchev Thank you for your review 😊
Sorry I'm not good at English, so please clear my head 🙏  

1. Your feedback is adding the code to check the arity of `step_text`, right?
2. Does this `step_text` means the method overrode in Gradle action?

> but the alternative would require re-architecting these methods to not be class methods which also has some downsides.

3. Could you give me more details about this comment 🙇

Commit:
Removed extra params and checking arity and added tests

Commit:
Forgot to remove an unused param

Commit:
Rebased to include PR for :tasks and removed option to make this step text the only behavior

joshdholtz reviewed (approved):
I really love this feature! Thanks for doing this 😊
